ai/aitraits.cpp

```diff
@@ -33,8 +33,7 @@ void ai_traits_init(struct player *pplayer)
 {
   enum trait tr;
 
-  pplayer->ai_common.traits = static_cast<ai_trait *>(fc_realloc(
-      pplayer->ai_common.traits, sizeof(struct ai_trait) * TRAIT_COUNT));
+  pplayer->ai_common.traits = std::vector<ai_trait>(TRAIT_COUNT);
```

Comment:
This doesn't seem to be ever resized. `TRAIT_COUNT` is 3 and `ai_trait` is just two `int`s. Would this make more sense as a `std::array<ai_trait, TRAIT_COUNT>`?

Reply:
I was considering the same, also we have waay too much ram these days to have a lot more traits still in stack,  Ended up doing the same in heap because I thought the difference would be negligible, and array didnt have clear.